pkg/apis/serving/k8s_validation.go

```diff
@@ -267,7 +270,84 @@ func ValidatePodSpec(ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
+func validateContainers(ctx context.Context, containers []corev1.Container, volumes sets.String) *apis.FieldError {
+	var errs *apis.FieldError
+	cfg := config.FromContextOrDefaults(ctx).Defaults
+	if !cfg.EnableMultiContainer {
+		errs = errs.Also(apis.ErrMultipleOneOf("containers"))
```

Comment:
Do we want to change the message here to be along the lines of
"enable-multi-container is off, but %d containers found"?

Reply:
Do you mean removing existing one and adding something like this
```
errs = errs.Also(&apis.FieldError{
			Message: fmt.Sprintf("enable-multi-container is off, but found %d containers", len(containers)),
		})
```

Reply:
Yep.